### PR TITLE
fix: genesis auth account format

### PR DESCRIPTION
**Pull request by dudong2**
## Description

I restored the part(MarshalX, UnmarshalX) that was removed due to a build failure while bumping up to cosmos-sdk@v0.45.1.

wrong format
```json
{
  "@type": "/lbm.auth.v1.BaseAccount",
  "account_number": "0",
  "address": "link146asaycmtydq45kxc8evntqfgepagygelel00h",
  "ed25519_pub_key": null,
  "multisig_pub_key": null,
  "secp256k1_pub_key": {
      "key": "AgT2QPS4Eu6M+cfHeba+3tumsM/hNEBGdM7nRojSZRjF"
  },
  "secp256r1_pub_key": null,
  "sequence": "1"
},
```

right format
```json
{
  "@type": "/lbm.auth.v1.BaseAccount",
  "account_number": 0,
  "address": "link146asaycmtydq45kxc8evntqfgepagygelel00h",
  "pub_key": {
      "key": "eyJrZXkiOiJBZ1QyUVBTNEV1Nk0rY2ZIZWJhKzN0dW1zTS9oTkVCR2RNN25Sb2pTWlJqRiJ9",
      "type": 1
  },
  "sequence": "1"
},
```

closes: #515 

## Checklist:

- [ ] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/wasm/keeper/keeper_test.go [hunk 1]
```diff
@@ -319,7 +319,7 @@ func TestInstantiate(t *testing.T) {
 
 	gasAfter := ctx.GasMeter().GasConsumed()
 	if types.EnableGasVerification {
-		require.Equal(t, uint64(0x142fa), gasAfter-gasBefore)
+		require.Equal(t, uint64(0x1402a), gasAfter-gasBefore)
```
Comment (zemyblue) [inline x/wasm/keeper/keeper_test.go:322]: Why is the fee reduced?

**Comment by zemyblue** [comment]
The unittest of vestingAccount's marchaling and unmarshaling functions were not existed. Please add it and increase the coverage.

File: x/wasm/keeper/keeper_test.go (hunk 1 above)
Comment (dudong2) [reply x/wasm/keeper/keeper_test.go:322]: I think the cause is the decrease in the size of msg as pubkey is distinguished by type.

**Review by zemyblue — APPROVED** [review]
LGTM

File: x/auth/types/account_test.go [hunk 2]
```diff
@@ -214,6 +215,28 @@ func TestModuleAccountJSON(t *testing.T) {
 	require.Equal(t, acc.String(), a.String())
 }
 
+func TestModuleAccountJSONPB(t *testing.T) {
+	pubkey := secp256k1.GenPrivKey().PubKey()
+	addr := sdk.BytesToAccAddress(pubkey.Address())
+	baseAcc := types.NewBaseAccount(addr, nil, 10, 50)
+	acc := types.NewModuleAccount(baseAcc, "test", "burner")
```
Comment (0Tech) [inline x/auth/types/account_test.go:222]:
```suggestion
	acc := types.NewModuleAccount(baseAcc, "test", types.Burner)
```
How about using `types.Burner`?

File: x/auth/types/account_test.go [hunk 3]
```diff
@@ -214,6 +215,28 @@ func TestModuleAccountJSON(t *testing.T) {
...
+	jum := jsonpb.Unmarshaler{}
+	addr2 := sdk.AccAddress("")
+	baseAcc2 := types.NewBaseAccount(addr2, nil, 0, 0)
+	acc2 := types.NewModuleAccount(baseAcc2, "")
+	err = acc2.UnmarshalJSONPB(&jum, bz)
+	require.NoError(t, err)
+
+	// error on bad bytes
+	err = acc2.UnmarshalJSONPB(&jum, bz[:len(bz)/2])
+	require.NotNil(t, err)
```
Comment (0Tech) [inline x/auth/types/account_test.go:237]:
```suggestion
	require.Error(t, err)
```
How about using `Error()`?

File: x/auth/vesting/types/vesting_account_test.go [hunk 4]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+	acc := types.NewBaseVestingAccount(baseAcc, coins, time.Now().Unix())
+
+	jm := jsonpb.Marshaler{}
+	bz, err := acc.MarshalJSONPB(&jm)
+	require.Nil(t, err)
+
+	jum := jsonpb.Unmarshaler{}
+	acc2 := types.NewBaseVestingAccount(nil, sdk.NewCoins(), 0)
+	err = acc2.UnmarshalJSONPB(&jum, bz)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:818]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 5]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+	require.IsType(t, &types.BaseVestingAccount{}, acc2)
+	require.Equal(t, acc.String(), acc2.String())
+
+	// error on bad bytes
+	err = acc2.UnmarshalJSONPB(&jum, bz[:len(bz)/2])
+	require.NotNil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:824]:
```suggestion
	require.Error(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 6]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+}
+
+func TestContinuousVestingAccountMarshalJSONPB(t *testing.T) {
+	baseAcc, coins := initBaseAccount()
+	baseVesting := types.NewBaseVestingAccount(baseAcc, coins, time.Now().Unix())
+	acc := types.NewContinuousVestingAccountRaw(baseVesting, baseVesting.EndTime)
+
+	jm := jsonpb.Marshaler{}
+	bz, err := acc.MarshalJSONPB(&jm)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:834]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 7]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+
+	jum := jsonpb.Unmarshaler{}
+	baseVesting2 := types.NewBaseVestingAccount(nil, sdk.NewCoins(), 0)
+	acc2 := types.NewContinuousVestingAccountRaw(baseVesting2, baseVesting2.EndTime)
+	err = acc2.UnmarshalJSONPB(&jum, bz)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:840]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 8]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+}
+
+func TestPeriodicVestingAccountMarshalJSONPB(t *testing.T) {
+	baseAcc, coins := initBaseAccount()
+	baseVesting := types.NewBaseVestingAccount(baseAcc, coins, time.Now().Unix())
+	acc := types.NewPeriodicVestingAccountRaw(baseVesting, baseVesting.EndTime, types.Periods{types.Period{3600, coins}})
+
+	jm := jsonpb.Marshaler{}
+	bz, err := acc.MarshalJSONPB(&jm)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:856]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 9]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+
+	jum := jsonpb.Unmarshaler{}
+	baseVesting2 := types.NewBaseVestingAccount(nil, sdk.NewCoins(), 0)
+	acc2 := types.NewPeriodicVestingAccountRaw(baseVesting2, baseVesting2.EndTime, types.Periods{})
+	err = acc2.UnmarshalJSONPB(&jum, bz)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:862]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 10]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+	require.IsType(t, &types.PeriodicVestingAccount{}, acc2)
+	require.Equal(t, acc.String(), acc2.String())
+
+	// error on bad bytes
+	err = acc2.UnmarshalJSONPB(&jum, bz[:len(bz)/2])
+	require.NotNil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:868]:
```suggestion
	require.Error(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 11]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+}
+
+func TestDelayedVestingAccountMarshalJSONPB(t *testing.T) {
+	baseAcc, coins := initBaseAccount()
+	acc := types.NewDelayedVestingAccount(baseAcc, coins, time.Now().Unix())
+
+	jm := jsonpb.Marshaler{}
+	bz, err := acc.MarshalJSONPB(&jm)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:877]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 12]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+
+	jum := jsonpb.Unmarshaler{}
+	acc2 := types.NewDelayedVestingAccount(nil, sdk.NewCoins(), 0)
+	err = acc2.UnmarshalJSONPB(&jum, bz)
+	require.Nil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:882]:
```suggestion
	require.NoError(t, err)
```

File: x/auth/vesting/types/vesting_account_test.go [hunk 13]
```diff
@@ -803,6 +804,90 @@ func TestPermanentLockedAccountMarshal(t *testing.T) {
...
+	require.IsType(t, &types.DelayedVestingAccount{}, acc2)
+	require.Equal(t, acc.String(), acc2.String())
+
+	// error on bad bytes
+	err = acc2.UnmarshalJSONPB(&jum, bz[:len(bz)/2])
+	require.NotNil(t, err)
```
Comment (0Tech) [inline x/auth/vesting/types/vesting_account_test.go:888]:
```suggestion
	require.Error(t, err)
```

**Review by 0Tech — COMMENTED** [review]
LGTM, but some improvements.

**Review by 0Tech — APPROVED** [review]

**Pull request merged**